src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -486,6 +494,242 @@ ov::npuw::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
+    std::vector<std::size_t> cpu_closure_ids;
+    for (std::size_t cidx = 0; cidx < closure.size(); ++cidx) {
+        if (closure_uid[cidx] == -1) {  // CPU closure, not in the bank
+            cpu_closure_ids.push_back(cidx);
+            cpu_closures.push_back(closure[cidx]);
+        }
+    }
+
+    write(stream, cpu_closure_ids);
+
+    for (const auto& tensor : cpu_closures) {
+        write(stream, tensor);
+    }
+
+    // FIXME: support weightless flow!
+
+    LOG_DEBUG("DONE.");
+}
+
+ov::npuw::CompiledModel::CompiledModelDesc ov::npuw::CompiledModel::CompiledModelDesc::deserialize(
+    std::istream& stream,
+    const std::size_t idx,
+    const std::shared_ptr<const ov::IPlugin>& plugin,
+    const ov::AnyMap& properties) {
+    using namespace ov::npuw::s11n;
+
+    LOG_DEBUG("Deserializing CompiledModelDesc...");
+    LOG_BLOCK();
+
+    CompiledModelDesc desc;
+
+    std::string device;
+    read(stream, device);
+    read(stream, desc.replaced_by);
+    // FIXME: is this check enough?
+    if (desc.replaced_by == idx) {
+        // Import model from either NPU or CPU plugin
+        // FIXME: workaround for import/export model since import model seems to reset the file pointer
+        std::string buf;
+        read(stream, buf);
+
+        // FIXME: extra copy
+        std::stringstream buffer;
+        buffer.write(&buf[0], buf.size());
+
+        // No NPUW properties are present in this config
+        desc.compiled_model = plugin->get_core()->import_model(buffer, device, properties);
+    }
+
+    read(stream, desc.param_base);
+    read(stream, desc.forced_to_fcall);
+
+    read(stream, desc.host_gather.dst_idx);
+    read(stream, desc.host_gather.src_idx);
+    read(stream, desc.host_gather.idx_idx);
+
+    read(stream, desc.spatial);
+
+    read(stream, desc.scales);
+    read(stream, desc.zerops);
+    read(stream, desc.is_remote);
+
+    // NOTE: for closure only deserialize uids - full flow
+    read(stream, desc.closure_uid);
+
+    // Some tensors might be present in CPU closure already - need to deserialize as is
+    // FIXME: When weightless serialization is introduced, this should be handled differently
+    std::size_t closure_size = 0;
+    read(stream, closure_size);
+    std::vector<std::size_t> cpu_closure_ids;
+    read(stream, cpu_closure_ids);
+    desc.closure.resize(closure_size);
+    for (const auto& cidx : cpu_closure_ids) {
+        read(stream, desc.closure[cidx]);
+    }
+
+    // FIXME: support weightless flow!
+
+    LOG_DEBUG("DONE.");
+
+    return desc;
+}
+
+void ov::npuw::CompiledModel::serialize(std::ostream& stream) const {
+    LOG_INFO("Serializing CompiledModel...");
+    LOG_BLOCK();
+
+    using namespace ov::npuw::s11n;
+
+    // Serialize name first + NPUW identifier for deserialization
+    // FIXME: consider a better approach then using name there
+    std::string name = std::string(NPUW_NAME_IDENTIFIER) + m_name;
+    write(stream, name);
```

Comment:
Why do we need some identifier for deserialization?

Reply:
In constructor of CompiledModel and LLMCompiledModel I somehow need to identify that it's being called from deserialize(). So I need some mechanism to distinguish default constructor behavior from a dummy deserialized one

Reply:
Done